src/Storages/PartitionedSink.cpp

```diff
@@ -51,16 +50,20 @@ SinkPtr PartitionedSink::getSinkForPartitionKey(StringRef partition_key)
     return it->second;
 }
 
-void PartitionedSink::consume(Chunk & chunk)
+void PartitionedSink::consume(Chunk & input_chunk)
 {
-    const auto & columns = chunk.getColumns();
-
     Block block_with_partition_by_expr = sample_block.cloneWithoutColumns();
-    block_with_partition_by_expr.setColumns(columns);
+    block_with_partition_by_expr.setColumns(input_chunk.getColumns());
     partition_by_expr->execute(block_with_partition_by_expr);
 
     const auto * partition_by_result_column = block_with_partition_by_expr.getByName(partition_by_column_name).column.get();
 
+    /*
+     * `write_partition_columns_into_files`
+     */
+    const auto chunk = getPartitionStrategy()->getChunkWithoutPartitionColumnsIfNeeded(input_chunk);
```

Comment:
why not just `partition_strategy` here?
```suggestion
    const auto chunk = partition_strategy->getChunkWithoutPartitionColumnsIfNeeded(input_chunk);
```